src/specfem3D/write_seismograms.f90

```diff
@@ -250,6 +253,48 @@ subroutine write_seismograms_to_file()
     call close_asdf_data()
   endif
 
+  ! write 3D seismogram array
+  if (OUTPUT_SEISMOS_3D_ARRAY) then
+    write(sisname,'(A,I5.5)') '/array_seismograms_node_',myrank
+    write(staname,'(A,I5.5)') '/array_stations_node_',myrank
+    if (seismo_offset == 0) then
+      open(unit=IOUT,file=trim(OUTPUT_FILES)//trim(sisname)//'.bin',status='unknown',form='unformatted',action='write')
+    else
+      open(unit=IOUT,file=trim(OUTPUT_FILES)//trim(sisname)//'.bin',status='old', &
+            form='unformatted',position='append',action='write')
+    endif
+    write(IOUT) seismograms
+    close(IOUT)
+    ! save list of stations in current processor
+    if (seismo_offset == 0) then
+      if (myrank == 0) then
+        open(unit=IOUT,file=trim(OUTPUT_FILES)//'seismogram_stats.txt',status='unknown',form='formatted',action='write')
+        write(IOUT,*) 'DT    =', DT
+        write(IOUT,*) 'NSTEP =', seismo_current
+      endif
+      open(unit=IOUT,file=trim(OUTPUT_FILES)//trim(staname)//'.txt',status='unknown',form='formatted',action='write')
```

Comment:
for myrank == 0, this opens another file with the same unit number as the file just above, therefore becomes ambiguous. it would need another unit number or the file above should be closed first.

Reply:
please fix the above in another PR. I'm going to merge this one into devel for now.